doc/tutorials/acrn-dm_QoS.rst

```diff
@@ -0,0 +1,119 @@
+.. _acrn-dm_qos:
+
+Enable QoS based on runC container
+##################################
+ACRN supports Device-Model QoS based on runC container to control
+the SOS resources (CPU, Storage, MEM, NET) by modifying the runC configuration file.
+
+What is QoS
+***********
+Traditionally, Quality of Service (QoS) is the description or measurement
+of the overall performance of a service, such as a `computer network
+<https://en.wikipedia.org/wiki/Computer_network>`_ or a `cloud computing
+<https://en.wikipedia.org/wiki/Cloud_computing>`_ service,
+particularly the performance is seen by the users on the network.
+
+What is runC container
+**********************
+Containers are an abstraction at the application layer that packages code
+and dependencies together. Multiple containers can run on the same machine
+and share the OS kernel with other containers, each running as
+isolated processes in user space. runC, a lightweight universal container runtime,
+is a command-line tool for spawning and running containers according
+to the Open Container Initiative (OCI) specification.
+
+ACRN-DM QoS architecture
+************************
+In ACRN-DM QoS design, we run the ACRN-DM in a runC container environment.
+Every time we start a UOS, we run a runC container at first;
+then launch the ACRN-DM in the started container. The resources
+the ACRN-DM QoS can manage for Device-Model:
+
+- CPU utilization
+- Memory amount/limitation
+- I/O bandwidth
+- Network throughput
+
+.. figure:: images/acrn-dm_qos_architecture.png
+   :align: center
+
+   ACRN-DM QoS architecture
+
+ACRN-QoS CPU utilization example
+********************************
+In runC ``config.json`` we set the CPU resource as below for VM0 and VM1:
+
+.. code-block:: none
+
+    "cpu": {
+       "shares": 1024,
+       "quota": 1000000,
+       "period": 500000,
+       "realtimeRuntime": 950000,
+       "realtimePeriod": 1000000,
+       "cpus": "2-3",
+       "mems": "0-7"
+    },
+
+In this example the CPU shares value is 1024, so the VM0 or VM1 device model
+CPU utilization is ``1024 / (1024 + 1024 +1024) = 33%``, which means
+the maximal CPU resource for the VM0 or VM1 is 33% of the entire CPU resource.
+
+.. figure:: images/cpu_utilization_image.png
+   :align: center
+
+   CPU utilization image
+
+How to use ACRN-DM QoS
+**********************
+
+ #. Add these parameters to the ``runC.json`` file:
```

Comment:
Where should this file be placed? Is there a set of standard locations, or should it specifically be put in the same folder as the launch script for example?

Reply:
Yes, this configuration have been set to the same place with the launch script, we we run the launch script it will be copy to the /usr/share/acrn/conf directory.

Reply:
When you say "run the launch script", do you mean when you run `acrnctl add`?

Can  you amend the text here to indicate the `runC.json` should be placed in the working directory?

Is there any risk to run into name conflicts if you repeat these operations for a few VMs?

Reply:
> When you say "run the launch script", do you mean when you run `acrnctl add`?

Yes, I mean add the launch script to the `acrnctl`

> Can you amend the text here to indicate the `runC.json` should be placed in the working directory?

Sure, @Ragnarliu will indicate the full directory of runC.json.

> Is there any risk to run into name conflicts if you repeat these operations for a few VMs?

Right now the launch_uos hard code the VM name, you can get this from launch_clear function parameter.
If you want to start more VM you need to modify the launch_uos script.